appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/presenter/AppsDiscoveryListPresenter.kt

```diff
@@ -8,6 +8,18 @@ import org.kinecosystem.appsdiscovery.sender.repositories.OperationResultCallbac
 
 class AppsDiscoveryListPresenter(private val discoveryAppsRepository: DiscoveryAppsRepository) : BasePresenter<IAppsDiscoveryListView>(), IAppsDiscoveryListPresenter {
 
+    private var loadingListener:LoadingListener? = null
+
+    interface LoadingListener{
+        fun loading()
+        fun loadingComplete()
+        fun loadingFailed()
+    }
+
+    override fun setLoadingListener(loadingListener: LoadingListener) {
+        this.loadingListener = loadingListener
```

Comment:
you need to make sure the listener is set to null when the activity is destroyed

Reply:
👍

Reply:
on detach i  reset it to null